### PR TITLE
feat/user management

### PR DESCRIPTION
**Pull request by Kamleshpaul**
https://github.com/brendt/rfc-vote/issues/175

### CODE REVIEW

**Comment by Plytas** [comment]
Would be nice if you could add tests for this.

**Comment by Kamleshpaul** [comment]
> Would be nice if you could add tests for this.

i m working on it

**Comment by Kamleshpaul** [comment]
@Plytas tests is added. 
Now in my side it is done and ready to merge let me know if anything need to change

File: app/Http/Controllers/UserEditController.php
Comment (Plytas) [inline app/Http/Controllers/UserEditController.php:27]: This path is not covered by tests. Could you add one that does?

File: tests/Feature/AdminUserTest.php
Comment (Plytas) [inline tests/Feature/AdminUserTest.php:60]: If `$this->faker->boolean` returns `false` then test fails.

File: app/Http/Controllers/UserEditController.php
Comment (Plytas) [inline app/Http/Controllers/UserEditController.php:8]:
```suggestion
use Illuminate\Contracts\View\View;
```

File: app/Http/Livewire/UserList.php
Comment (Plytas) [inline app/Http/Livewire/UserList.php:6]:
```suggestion
use Illuminate\Contracts\View\View;
```

**Review by Plytas — CHANGES_REQUESTED** [review]

**Comment by Kamleshpaul** [comment]
@Plytas  fixed changes

**Comment by joshbonnick** [comment]
@brendt please check your email or twitter DMs before merging this PR. Slightly unrelated issue that will be made worse by merging this before fixing separate issue.

File: app/Http/Controllers/UserEditController.php
Comment (brendt) [inline app/Http/Controllers/UserEditController.php:27]: Could you use the `flash('Success');` helper function everywhere you need a flash message?

File: resources/views/livewire/user-list.blade.php
Comment (brendt) [inline resources/views/livewire/user-list.blade.php:19]: @SerhiiCho would you want to take a look from the frontend point of view? We don't need to address issues in this PR necessarily, but it might be good trying to settle on reusable admin components.

Comment (SerhiiCho) [reply resources/views/livewire/user-list.blade.php:19]: Sorry, but this PR has no description, I have no idea what is going on here

Comment (brendt) [reply resources/views/livewire/user-list.blade.php:19]: It's addressing this issue: https://github.com/brendt/rfc-vote/issues/175

No worries if you don't have the time for it :)

File: app/Http/Controllers/UserEditController.php
Comment (Kamleshpaul) [reply app/Http/Controllers/UserEditController.php:27]: okay sure

File: resources/views/livewire/user-list.blade.php
Comment (SerhiiCho) [reply resources/views/livewire/user-list.blade.php:12]: I have time, I just don't know how to pull this to my computer and push changes since it's separate origin

Comment (brendt) [reply resources/views/livewire/user-list.blade.php:12]: Gotcha! Github actually shows you the commands:

Screenshot 2023-09-06 at 20 43 11

Press "Command line instructions":

Screenshot 2023-09-06 at 20 43 17

**Comment by brendt** [comment]
Thanks, appreciate it! Let's wait a bit for Sehrii's feedback, I'm good to merge from my side :) 

Again: we don't need to fix frontend stuff within this PR, just want to make sure we don't miss anything before merging :)

**Review by SerhiiCho — COMMENTED** [review]
Looks good, unfortunately, I have only read-only access, I cannot propose any changes for some reason

**Review by SerhiiCho — APPROVED** [review]

**Comment by SerhiiCho** [comment]
Wow! I figured out how to do that, to make changes in Kamlesh's branch, I need:
- to set a separate remote origin as his repo
- pull it to my computer
- make changes
- make PR to his dev branch

I gotta learn git and github more 🧠

File: resources/views/livewire/user-list.blade.php
Comment (SerhiiCho) [reply resources/views/livewire/user-list.blade.php:12]: It's easier for you because you are the owner, for me, it's a bit tricky (i guess)

**Comment by brendt** [comment]
Now I'm the GitHub noob, because I'm not sure whether your suggestions were actually merged or not 🤔

Btw @SerhiiCho would you mind reaching out via email? I wanted to discuss write permissions, but have no way of reaching you, no I'm just hijacking this comment 😅

**Comment by SerhiiCho** [comment]
Merge it if you want, I just formatted one file. There was nothing critical there

**Comment by brendt** [comment]
Great! Btw, forgot to add my email >< 

`brent.roose@jetbrains.com`

**Pull request merged**
